managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpAddEntry.java

```diff
@@ -176,7 +186,7 @@ public void addComplete(int rc, final LedgerHandle lh, long entryId, Object ctx)
         if (!STATE_UPDATER.compareAndSet(OpAddEntry.this, State.INITIATED, State.COMPLETED)) {
             log.warn("[{}] The add op is terminal legacy callback for entry {}-{} adding.", ml.getName(), lh.getId(),
                     entryId);
-            OpAddEntry.this.recycle();
+            OpAddEntry.this.recycleAfterClosed();
```

Comment:
Instead of adding `recycleAfterClosed`, let's just skip recycling in cases that cause problems.

Reply:
> Instead of adding recycleAfterClosed, let's just skip recycling in cases that cause problems.

Good idea. Done